### ISSUE TITLE
virtual hosting scenario gives NotFound error

### ISSUE THREAD

**Issue by tomgross** [plone/plone.rest#48]
With plone.rest (and https://github.com/plone/plone.restapi) installed accessing the content via VHM does not work.

## My Setup is

Plone 5.0.2
plone.rest (rev=e6c686bb42f89ab64df113bc1084ff09a0e2ac74)
plone.restapi (rev=503de8767de6d509b36974e9c91b6324c6d9a59a)

## Steps to reproduce
`curl  -X GET --header "Accept: application/json" http://localhost:8080/VirtualHostBase/http/localhost:8080/Plone/VirtualHostRoot/`

Gives
`{
  "message": "Resource not found: http://localhost:8080/Plone/virtual_hosting", 
  "type": "NotFound"
}`

The expected output is
```
{
  "@context": "http://www.w3.org/ns/hydra/context.jsonld", 
  "@id": "http://localhost:8000/Plone", 
  "@type": "Plone Site", 
  "member": [
    {
      "@id": "http://localhost:8000/Plone/front-page", 
      "description": "Congratulations! You have successfully installed Plone.", 
      "title": "Welcome to Plone"
    }, 
    {
      "@id": "http://localhost:8000/Plone/news", 
      "description": "Site News", 
      "title": "News"
    }, 
    {
      "@id": "http://localhost:8000/Plone/events", 
      "description": "Site Events", 
      "title": "Events"
    }, 
    {
      "@id": "http://localhost:8000/Plone/Members", 
      "description": "Site Users", 
      "title": "Users"
    }, 
...
  ], 
  "parent": {}
}

```
which is correctly produced **without** virtual hosting component:
`curl --X GET --header "Accept: application/json" http://localhost:8000/Plone/`

**Comment by tomgross** [comment]
The VHM is filtered out here:
https://github.com/plone/plone.rest/blob/master/src/plone/rest/traverse.py#L19

**Comment by buchi** [comment]
Fixed in #49
